src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2164,6 +2167,13 @@ var projectUniqueNamesForBuildIntToUpdate
                         }
                     }
 
+                    if (msbuildProject != null)
+                    {
+                        // end batch for msbuild and let it save everything.
+                        // always calls it before PostProcessAsync or binding redirects
+                        msbuildProject.ProjectSystem.EndProcessing();
```

Comment:
What is the impact of failing to balance a `BeginProcessing()` call with an `EndProcessing()` call?  For example, suppose one of the method calls between `BeginProcessing()` and `EndProcessing()` above threw an exception, then this statement would not execute.  Should this be called in a `finally` block?

This PR removes from [src/NuGet.Core/NuGet.PackageManagement/Projects/MSBuildNuGetProject.cs](https://github.com/NuGet/NuGet.Client/pull/1435/files#diff-4e7dc9aeba96ffd6de20e4390a27a1f4) calls to `BeginProcessing()` and `EndProcessing()`.  However, before the removal of those calls `EndProcessing()` was always called in a `finally` block.